pkg/ccl/changefeedccl/helpers_test.go

```diff
@@ -143,8 +143,10 @@ func applySourceAssertion(
 	}
 	for _, m := range payloads {
 		var message map[string]any
-		if err := gojson.Unmarshal(m.Value, &message); err != nil {
-			return errors.Wrapf(err, `unmarshal: %s`, m.Value)
+		decoder := gojson.NewDecoder(bytes.NewReader(m.Value))
+		decoder.UseNumber()
+		if err := decoder.Decode(&message); err != nil {
+			return errors.Wrapf(err, `decode: %s`, m.Value)
```

Comment:
you need to also check that decoder has read all the bytes in the input. one way to do this is to compare https://pkg.go.dev/encoding/json#Decoder.InputOffset with len(m.Value)